SIMSBIOHUB-515: API Clean up

KjartanE opened the pull request:
## Links to Jira Tickets

- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-515

## Description of Changes
- Remove unused code
    - Transformation Schemas
    - Validation Schemas
    - DWC
        - Endpoints
        - Service
        - Repository
        - Frontend components
        - Hooks
    - Summary
        - Endpoints
        - Service
        - Repository
        - Frontend dataloader
        - Hooks
    - XLSX (Note: removed old code, current xlsx is still functional)
        - Endpoints
        - Validation Service
        - Hooks
    - Project Publish Metadata
        - Endpoints
        - history publish service and repo functions
        - Hooks

- Added `additionalProperties: false` to all `type: 'object'` Openapi schemas
    - Critter
    - GeoJSON
    - pagination
    - Project
        - Attachments
        - Participant
        - Survey
            - Attachments
            - Critters
            - Observations
            - Sample Site
            - Telemetry
            - Publish
    - Administrative Activity
    - Codes
    - Funding Sources
    - GCNotify
    - Regions
    - Users
    - Roles

- Created Common Openapi Schemas
    - `User`
        - `ProjectUser`
        - `SystemUser`
        - `projectAndSystemUserSchema`
        - `surveyParticipationAndSystemUserSchema`

## Testing Notes

- {List any relevant testing considerations, necessary pre-reqs, and areas of the app to focus on. Specifically, include anything that will help the reviewers of this PR verify the code is functioning as expected.}

KjartanE committed:
initial update for openapi schema, wip

KjartanE committed:
remove dwc endpoints, wip

KjartanE committed:
remove summary/xlsx/validation unused files. wip

KjartanE committed:
history publish for projects removed

curtisupshall committed:
SIMSBIOHUB-517, 518: Bug fixes (#1232)

* SIMSBIOHUB-517: Fixed bug where form error snackbar appeared after errors were amended
* SIMSBIOHUB-518: Fix validation error persistence bug when selecting focal species

NickPhura committed:
TechDebt: Fix Deployed App Config Not Matching Local Config (#1234)

* Fix deployed app config not matching local config
* Install missing 'request' library

KjartanE committed:
Fix apiSchema for users/Funding/Project Pages

KjartanE committed:
Merge branch 'dev' into SIMSBIOHUB-515

NickPhura commented:
@KjartanE If it helps, I think we can probably get rid of the api unit tests that were testing the endpoint openapi specs. It kind of feels like overkill, and with the additionalProperties field set, there should be little or no risk of it being incorrect going forward.

KjartanE committed:
Project update fixxed

curtisupshall committed:
Tech debt: create the database procedures folder (#1233)

SIMSBIOHUB-TechDebt: Project/ Survey delete procedure can now be updated without a migration. Misc text changes/ UI updates

---------

Co-authored-by: Nick Phura <Nick.Phura@quartech.com>

KjartanE committed:
fix project attachments endpoints

KjartanE committed:
survey schema objects

curtisupshall committed:
SIMSBIOHUB-492: Fix project advanced filters (#1235)

* Fixes the projects list advanced filters to support paginated results with advanced filters applied and ITIS TSN searching.
* Added support to search on multiple species.
* Used MUI Collapse to show/hide advanced filters form.
* Code cleanup, minor tech debt.

KjartanE committed:
survey edit cleanup

al-rosenthal committed:
SIMSBIOHUB-490/521: Import Observations with measurements (#1227)

Updates to survey observation api/app components:
- Support import of Critterbase measurement columns in observation CSV.
- Support add/edit/view of Critterbase measurement columns in observation table.
- Misc tech debt, enhancements, improvements.

---------

Co-authored-by: Nick Phura <nickphura@gmail.com>
Co-authored-by: jeznorth <twenty15creative@gmail.com>

al-rosenthal committed:
SIMSBIOHUB-490/521: Import Observations with measurements (#1227)

Updates to survey observation api/app components:
- Support import of Critterbase measurement columns in observation CSV.
- Support add/edit/view of Critterbase measurement columns in observation table.
- Misc tech debt, enhancements, improvements.

---------

Co-authored-by: Nick Phura <nickphura@gmail.com>
Co-authored-by: jeznorth <twenty15creative@gmail.com>

KjartanE committed:
Merge branch 'dev' into SIMSBIOHUB-515

KjartanE committed:
survey create update

jeznorth committed:
Front-end Clean-up (#1239)

Front-end clean-up on the following interfaces:

* Taxonomy Displays
* Project team members list (autocomplete)
* Project List Page filters
* General Layout Updates / Formatting

---------

Co-authored-by: Nick Phura <nickphura@gmail.com>
Co-authored-by: Alfred Rosenthal <alfred.k.rosenthal@gmail.com>

NickPhura committed:
Fix some endpoints

NickPhura committed:
Fix self endpoint when user is admin on someone elses project (not a project participant).

mauberti-bc committed:
Survey Progress Field (#1238)

* survey repository changes & frontend

* openApi spec

* more openapi spec changes

* migration fixes

* wip: fix migration where survey table does not have progress_id

* set survey status when creating or editing surveys

* cleanup

* include description in planning ui options

* planning chip on survey page

* add survey progress chip to survey list page

* adjust colors of progress chip

* add survey_progress to tests

* linter

* cleanup and fixes

* linter

* make progress required

* wip: validation not yet working on survey form for progress component

* include progress in yup schema

* create index for progress id in survey table

* linter

* fix incorrect name: progress -> progress_id

* cleanup

* address PR comments

* update seed

* update comments in codes repository

* update returns types in codes repository

---------

Co-authored-by: Al Rosenthal <alfred.k.rosenthal@gmail.com>
Co-authored-by: Nick Phura <Nick.Phura@quartech.com>

NickPhura committed:
Merge remote-tracking branch 'origin/dev' into SIMSBIOHUB-515

KjartanE committed:
Merge remote-tracking branch 'origin/dev' into SIMSBIOHUB-515

NickPhura committed:
Add back region insert/update in survey service

NickPhura committed:
Merge branch 'SIMSBIOHUB-515' of https://github.com/bcgov/biohubbc into SIMSBIOHUB-515

KjartanE committed:
fix tests

KjartanE committed:
Merge remote-tracking branch 'origin/dev' into SIMSBIOHUB-515

NickPhura committed:
Fix test

NickPhura committed:
Merge branch 'SIMSBIOHUB-515' of https://github.com/bcgov/biohubbc into SIMSBIOHUB-515

KjartanE committed:
format-lint fix

KjartanE committed:
fix sample site api spec

KjartanE committed:
create/update critter, update attachment

KjartanE committed:
sample-site block/stratums

NickPhura committed:
Merge branch 'dev' into SIMSBIOHUB-515

KjartanE committed:
Merge branch 'dev' into SIMSBIOHUB-515

NickPhura committed:
Fix bug with get observations query

KjartanE committed:
Merge branch 'dev' into SIMSBIOHUB-515

NickPhura committed:
Fix query

NickPhura committed:
Merge branch 'SIMSBIOHUB-515' of https://github.com/bcgov/biohubbc into SIMSBIOHUB-515

NickPhura committed:
Merge branch 'dev' into SIMSBIOHUB-515

curtisupshall reviewed (approved):
Everything appears to be working as expected
